upup/pkg/fi/ca.go

```diff
@@ -43,17 +44,30 @@ type KeystoreItem struct {
 	Data []byte
 }
 
+// Keyset is a parsed api.Keyset.
+type Keyset struct {
+	// LegacyFormat instructs a keypair task to convert a Legacy Keyset to the new Keyset API format.
+	LegacyFormat bool
+	Items        map[string]*KeysetItem
+	Primary      *KeysetItem
+}
+
+// KeysetItem is a certificate/key pair in a Keyset.
+type KeysetItem struct {
+	Id          string
+	Certificate *pki.Certificate
+	PrivateKey  *pki.PrivateKey
+}
+
 // Keystore contains just the functions we need to issue keypairs, not to list / manage them
 type Keystore interface {
-	// FindKeypair finds a cert & private key, returning nil where either is not found
-	// (if the certificate is found but not keypair, that is not an error: only the cert will be returned).
-	// This func returns a cert, private key and a bool.  The bool value is whether the keypair is stored
-	// in a legacy format. This bool is used by a keypair
-	// task to convert a Legacy Keypair to the new Keypair API format.
-	FindKeypair(name string) (*pki.Certificate, *pki.PrivateKey, bool, error)
+	pki.Keystore
 
-	// StoreKeypair writes the keypair to the store
-	StoreKeypair(id string, cert *pki.Certificate, privateKey *pki.PrivateKey) error
+	// FindKeyset finds a Keyset.
+	FindKeyset(name string) (*Keyset, error)
+
+	// StoreKeypair writes the keyset to the store.
+	StoreKeypair(id string, keyset *Keyset) error
```

Comment:
This is meant to overwrite.

The term "keypair" is unfortunately exposed through the kops CLI. A subsequent PR removes the term from `kops create secret`. The remaining references are in output and documentation so it can probably be changed to "keyset" as appropriate.

Reply:
Renamed.